src/libcore/iter/mod.rs

```diff
@@ -1954,6 +2253,23 @@ impl<I> Iterator for Take<I> where I: Iterator{
 
         (lower, upper)
     }
+
+    #[inline]
+    fn try_fold<Acc, Fold, R>(&mut self, init: Acc, mut fold: Fold) -> R where
+        Self: Sized, Fold: FnMut(Acc, Self::Item) -> R, R: Try<Ok=Acc>
+    {
+        if self.n == 0 {
+            Try::from_ok(init)
+        } else {
+            let n = &mut self.n;
+            self.iter.try_fold(init, move |acc, x| {
+                *n -= 1;
+                let r = fold(acc, x);
+                if *n == 0 { SearchResult::Found(r) }
+                else { SearchResult::from_try(r) }
+            }).into_try()
+        }
```

Comment:
It wasn't obvious to me if this was correct, but I *think* it is? The inversion of using SearchResult is a bit confusing even if the intention is probably the opposite.

Reply:
Hmm, it got named that way because it was first written for find, but then I ended up using it in far more places than just that.  I wonder if `LoopResult` with `Break` and `Continue` would be better -- would turn `SearchResult::from_try` into `LoopResult::break_if_error` (or `continue_if_ok`)...

Reply:
I renamed SearchResult, and I think I like this way better.  Thoughts?
https://github.com/scottmcm/rust/compare/iter-try-fold...scottmcm:iter-try-fold-experiment